flink-table-store-core/src/main/java/org/apache/flink/table/store/log/LogOptions.java

```diff
@@ -0,0 +1,192 @@
+                                    .linebreak()
+                                    .list(
+                                            formatEnumOption(LogConsistency.TRANSACTIONAL),
+                                            formatEnumOption(LogConsistency.EVENTUAL))
+                                    .build());
+
+    public static final ConfigOption<LogChangelogMode> CHANGELOG_MODE =
+            ConfigOptions.key("changelog-mode")
+                    .enumType(LogChangelogMode.class)
+                    .defaultValue(LogChangelogMode.AUTO)
+                    .withDescription(
+                            Description.builder()
+                                    .text("Specifies the log changelog mode for table.")
+                                    .linebreak()
+                                    .list(
+                                            formatEnumOption(LogChangelogMode.AUTO),
+                                            formatEnumOption(LogChangelogMode.ALL),
+                                            formatEnumOption(LogChangelogMode.UPSERT))
+                                    .build());
+
+    public static final ConfigOption<String> KEY_FORMAT =
+            ConfigOptions.key("key.format")
+                    .stringType()
+                    .defaultValue("json")
+                    .withDescription(
+                            "Specifies the key message format of log system with primary key.");
+
+    public static final ConfigOption<String> FORMAT =
+            ConfigOptions.key("format")
+                    .stringType()
+                    .defaultValue("debezium-json")
+                    .withDescription("Specifies the message format of log system.");
+
+    /** Specifies the startup mode for log consumer. */
+    public enum LogStartupMode implements DescribedEnum {
+        FULL(
+                "full",
+                "Performs a snapshot on the table upon first startup,"
+                        + " and continue to read the latest changes."),
+
+        LATEST("latest", "Start from the latest."),
+
+        FROM_TIMESTAMP("from-timestamp", "Start from user-supplied timestamp.");
+
+        private final String value;
+        private final String description;
+
+        LogStartupMode(String value, String description) {
+            this.value = value;
+            this.description = description;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+
+        @Override
+        public InlineElement getDescription() {
+            return text(description);
+        }
+    }
+
+    /** Specifies the log consistency mode for table. */
+    public enum LogConsistency implements DescribedEnum {
+        TRANSACTIONAL(
+                "transactional",
+                "only the data after the checkpoint can be seen by readers, the latency depends on checkpoint interval."),
```

Comment:
Nit: we'd better keep all descriptions consistent on the aspect of starting with an upper/lower-case.